src/store/channels/saga.ts

```diff
@@ -41,6 +42,30 @@ export function* markAllMessagesAsReadInChannel(action) {
   }
 }
 
+// mark all messages as read in current active channel (only if you're not in full screen mode)
+export function* markAllMessagesAsReadInCurrentChannel() {
+  const currentUser = yield select(currentUserSelector());
+  const isMessengerFullScreen = yield select((state) => state.layout.value.isMessengerFullScreen);
+  const activeChannelId = yield select((state) => state.chat.activeChannelId);
+  const activeChannelInfo = activeChannelId ? yield select(rawChannelSelector(activeChannelId)) : null;
+
+  // just ensure first that you're not in full screen mode before marking all messages as read in a "channel"
+  if (!isMessengerFullScreen && activeChannelId && activeChannelInfo?.unreadCount > 0) {
+    yield call(markAllMessagesAsRead, activeChannelId, currentUser.id);
+  }
+}
+
+// mark all messages in read in current active conversation
+export function* markAllMessagesAsReadInCurrentConversation() {
```

Comment:
Ah, I just realized these aren't as specific as I thought. This function should take the channel id and ONLY call `markAllMessagesAsRead` if the id passed in matches the active conversation. (and likewise for the channel one). Something like, `markUnreadIfActive(channelId)`

Reply:
Otherwise, we'll be marking the active channel as read every time anything loads in the background.

Reply:
:+1:, updated